### ISSUE TITLE
"Check Database" adds a link to each task - always.

### ISSUE THREAD

**Issue by abonhote** [eudennis/ultimate-todoist-sync-for-obsidian-experiment#8]
I think that one should be easy to fix. You're keeping the link information within the comment, like

```
%%[tid:: [12345](https://app.todoist.com/app/task/12345)]%% 
```

Whenever I click on "Check Database", a new link tag gets added. Even to the closed tasks!

Before:

```
- [ ] Test 📅 2024-11-27   #td %%[tid:: [12345](https://app.todoist.com/app/task/12345)]%% 
```

After:

```
- [ ] Test 📅 2024-11-27   #td [link](https://app.todoist.com/app/task/12345) %%[tid:: [12345](https://app.todoist.com/app/task/12345)]%% 
```

Very nasty taks to remove them ...

**Comment by eudennis** [comment]
hello there, thanks for reporting. It happens only with the "Check Database" button? I haven't touched on that one tbh, let me take a look

**Comment by abonhote** [comment]
> hello there, thanks for reporting. It happens only with the "Check Database" button? I haven't touched on that one tbh, let me take a look

Yeah, only if you check the database. I guess there's something hardcoded ...

**Comment by abonhote** [comment]
Line 289 of `settings.ts` goes through the md files and adds the link.

**Comment by eudennis** [comment]
I've looked at that and didn't find an easy way to reproduce it. This feature basically checks for the task ID in the cache and then looks for the tasks within the .md files to see if the tasks you get there have the link or not. 

Can you confirm which version of the plugin you are using? And have you been using this plugin for some time already?

Based on the example you gave, it might be messing up because the old tasks have a different link structure.

**Comment by abonhote** [comment]
Well, I created a new todoist account for that, along with a fresh obsidian vault, installed the plugin, configured it, created some tasks and was immediately able to reproduce it. V0.4.8.

**Comment by eudennis** [comment]
@abonhote just pushed 0.4.10, which I believe has fixed the issue. I found the issue after your last comment, and it now seems to be working as expected. Do you mind testing?

**Comment by abonhote** [comment]
Hi @eudennis

Wonderful, works perfectly fine. Thanks for the app uri scheme option 👍 

Cheers

André

**Comment by eudennis** [comment]
Awesome :) I will close here
